File: app/src/main/java/org/bochman/upapp/domain/interactor/SearchPlaceByQuery.java
package org.bochman.upapp.domain.interactor;

import org.bochman.upapp.data.repository.PoiRepository;

/**
 * Encapsulate a query search request.
 */
class SearchPlaceByQuery  {

    private final PoiRepository repo;

    SearchPlaceByQuery(PoiRepository repo){
        this.repo=repo;
    }

    void invoke(String query){
        repo.fetchPoiByQuery(query);
    }
}

File: app/src/main/java/org/bochman/upapp/domain/interactor/SearchPlaceByQuery.kt
package org.bochman.upapp.domain.interactor;

import com.google.android.gms.maps.model.LatLng
import org.bochman.upapp.data.repository.PoiRepository;

/**
 * Encapsulate a query search request.
 */
class SearchPlaceByQuery(private val poiRepository: PoiRepository){

    suspend operator fun invoke(query:String) =  poiRepository.fetchPoiByQuery(query)


}